bookkeeper-server/src/main/java/org/apache/bookkeeper/bookie/ScanAndCompareGarbageCollector.java

```diff
@@ -69,7 +66,7 @@
  *
  * <p>TODO: eliminate the direct usage of zookeeper here {@link https://github.com/apache/bookkeeper/issues/1331}
  */
-public class ScanAndCompareGarbageCollector implements GarbageCollector{
+public class ScanAndCompareGarbageCollector implements GarbageCollector, Supplier<Integer> {
```

Comment:
'Supplier' is not very clear, I see we have only one value to report, but maybe this is not a good reason.
What do you think?

Reply:
Please make a suggestion.

Reply:
ping @eolivelli

Reply:
Sorry, I missed the notification of your comment.

We can have a field Supplier<Integer> deletedLedgerCounterAccessor and link it to the stats, instead of making the whole class implement Supplier

Reply:
@eolivelli fixed

Reply:
ping @eolivelli